### PR TITLE
fix: Workaround Android 12 webview crash

### LINKED ISSUE #42514: Scrolling to the end of a post containing an Embed block crashes the Android native editor
## Description
When a long post contains an Embed block (e.g. Twitter, YouTube), the native editor will crash after scrolling to the end of the post. 

## Step-by-step reproduction instructions
1. Create a new post within the Android native editor. 
1. Copy a Twitter URL to your clipboard. (**Note:** [It cannot be a mobile Twitter URL.](https://core.trac.wordpress.org/ticket/54414))
1. Add a _Twitter_ block<sup>1</sup>. 
1. Dismissing the block settings bottom sheet to persist the Twitter URL for the block. 
1. Repeat steps 2-4 a few more times so that there enough content to scroll the first/last block out of view. 
1. Scroll the post content to the end causing the scrolling to cease. 

<sup>1</sup> Twitter block is one example, but the same outcome can occur for YouTube and other embeds. 

## Expected behaviour
The editor does not crash. 

## Actual behaviour
The editor crashes. 

<details><summary>Android Studio error logs</summary>

```
2022-07-18 14:57:45.103 2061-2346/org.wordpress.android.prealpha A/libc: Fatal signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0x20 in tid 2346 (RenderThread), pid 2061 (ndroid.prealpha)
2022-07-18 14:57:46.285 6630-6630/? E/USNET: USNET: appName: org.wordpress.android.prealpha
2022-07-18 14:57:46.675 6630-6709/org.wordpress.android.prealpha E/chromium: [ERROR:simple_file_enumerator.cc(21)] opendir /data/user/0/org.wordpress.android.prealpha/cache/webview_org.wordpress.android.prealpha/Default/HTTP Cache/Code Cache/js: No such file or directory (2)
2022-07-18 14:57:46.675 6630-6709/org.wordpress.android.prealpha E/chromium: [ERROR:simple_index_file.cc(617)] Could not reconstruct index from disk
```

</details>

## Screenshots or screen recording (optional)

https://user-images.githubusercontent.com/438664/179602388-d6cd7d5d-2d56-43e6-933d-82094835de60.mp4

## WordPress information
- WordPress version: 6.0
- Gutenberg version: 13.6.0
- Are all plugins except Gutenberg deactivated? Yes
- Are you using a default theme (e.g. Twenty Twenty-One)? Geologist

## Device information
- Device: Samsung Galaxy S20
- Operating system: Android 12
- WordPress app version: WP 20.2

### PR DESCRIPTION
**Pull request by dcalhoun**
## Related PRs

* https://github.com/wordpress-mobile/gutenberg-mobile/pull/5033

## What?
Fixes #42514. Apply a workaround to prevent a crash on Android when webviews are displayed within a scroll view. The crash appears to originate with incompatibility with Android 12's new overflow scroll behavior.

## Why?
The editor crash creates a bad user experience with the possibility of content loss.

## How?
Applying `opacity: 0.99` [prevents the crash](https://github.com/react-native-webview/react-native-webview/issues/1915#issuecomment-808869253) from occurring. I perceive this change to be less than ideal, and should be considered a temporary fix while we seek a more appropriate solution for the `react-native-webview` library.

I explored other workarounds like [`androidLayerType="software"`](https://github.com/react-native-webview/react-native-webview/issues/1915#issuecomment-880989194) as well, but feared possible performance issues from the change based upon https://github.com/react-native-webview/react-native-webview/issues/1915#issuecomment-1047605643 and my own research. However, I do not fully understand the `opacity: 0.99` workaround. It is possible that it similarly creates a "software layer."

## Testing Instructions
Perform the testing steps outlined in #42514 with various Embed variants (e.g. YouTube, Twitter) and posts of different content lengths.

## Screenshots or screencast 
n/a

### CODE REVIEW

**Review by derekblank — APPROVED** [review]
Thanks for the quick fix @dcalhoun. I can confirm the changes resolved the error, tested on Pixel 6 and Galaxy S20. LGTM. 🚀 

I did want to leave a note on how we might revisit the implementation and remove the workaround in the future. I'm fairly certain that using style and opacity fixes for functionality workarounds gives both of us pause, but after reviewing the [react-native-webview thread](https://github.com/react-native-webview/react-native-webview/issues/1915), there doesn't seem like a better solution, either. I had also tried one of the other solutions in the thread, like `androidLayerType="software"`, but that seems like just as much of a workaround.

> This change should be viewed as a temporary fix while we await a more appropriate solution for the react-native-webview library.

Based on your work researching the thread, do you feel the ultimate solution is to wait for react-native-webview to release a fix? If so, how might we best "keep up" with the issue resolution from react-native-webview's end? This comment is more directed at the future of these files, and how we might manage adding workaround changes.

**Comment by fluiddot** [comment]
Thanks @dcalhoun @derekblank for working on the fix 🎊. 

Following [Derek's comment](https://github.com/WordPress/gutenberg/pull/42517#pullrequestreview-1042896073), I checked [most commented issues in the `react-native-webview` repository](https://github.com/react-native-webview/react-native-webview/issues?q=is%3Aissue+is%3Aopen+sort%3Acomments-desc) and this one is referenced in position [`#1`](https://github.com/react-native-webview/react-native-webview/issues/429) and and [`#4`](https://github.com/react-native-webview/react-native-webview/issues/1915), so looks like it's a top one on that library. Based on what I've seen in the comments on both issues, looks like the workaround applied in this PR is solid and should be enough for the short/mid-term. For the long term, probably until it's fixed in the library we won't be able to make much progress on our side, although I'm concerned that the issue was spotted around 2 years ago and still there's no solution 🤔 .

**Comment by dcalhoun** [comment]
> I'm fairly certain that using style and opacity fixes for functionality workarounds gives both of us pause, but after reviewing the [react-native-webview thread](https://github.com/react-native-webview/react-native-webview/issues/1915), there doesn't seem like a better solution, either. 

@derekblank definitely. I should have more explicitly conveyed this in my PR description. 

> I had also tried one of the other solutions in the thread, like `androidLayerType="software"`, but that seems like just as much of a workaround.

Agreed. I explored this workaround as well, but feared possible performance issues from the change based upon [comments on the thread](https://github.com/react-native-webview/react-native-webview/issues/1915#issuecomment-1047605643) and my own research. However, I do not fully understand the `opacity: 0.99` workaround. It is possible that it similarly creates a "software layer." 

> Based on your work researching the thread, do you feel the ultimate solution is to wait for react-native-webview to release a fix? 

As @fluiddot [mentioned](https://github.com/WordPress/gutenberg/pull/42517#issuecomment-1188878886), it appears awaiting a solution from the `react-native-webview` maintainers may be a fruitless endeavor, given the age of the related issues. 

> If so, how might we best "keep up" with the issue resolution from react-native-webview's end? This comment is more directed at the future of these files, and how we might manage adding workaround changes.

Historically, my approach is subscribing to the relevant issues, and hoping that once it is resolved I remember what it is I need to do next after receiving a notification. Not the most sound approach. 🙃  

One approach to tracking workarounds like this would be [a single file implementing them](https://csswizardry.com/2013/04/shame-css/). The benefits would be a single, documented location that clearly conveys the intentions for a workaround and how we remove them in the future. 

---

> For the long term, probably until it's fixed in the library we won't be able to make much progress on our side, although I'm concerned that the issue was spotted around 2 years ago and still there's no solution 🤔 .

@fluiddot thank you for sharing the additional findings. The age of the issue is disconcerting. 😞 I agree it is unlikely for us to see a proper solution from the library. It may require that we investigate a proper solution ourselves.

**Comment by ParaskP7** [comment]
FYI: This fix made it into the [20.3-rc-2](https://github.com/wordpress-mobile/WordPress-Android/releases/tag/20.3-rc-2).

**Pull request merged**
